src/modals/vm-export/VmExportDataProvider.js

```diff
@@ -40,7 +40,7 @@ async function validateThinVm (vmId, sdId) {
   const vm = await engineGet(`api/vms/${vmId}`)
 
   // If a VM has both attributes set it's thin/dependent
-  if (vm.original_template.id === emptyGuid || vm.template.id === emptyGuid) {
+  if (vm.original_template?.id === emptyGuid || vm.template?.id === emptyGuid) {
```

Comment:
the original logic doesn't seem to be accurate since the vm is thin/dependent regardless of the value of the original-template field - if 'template' is set to a non-empty guid then the vm is thin/dependent, otherwise it's either cloned or based on the blank template (and then it doesn't matter if it's thin provisioned or cloned because the VM has no disk). but as it can't be that vm.template is not set to empty guid while vm.original_template is set to empty guid, it was kind of ok and the first part of the change looks alright

but as for the second part of this change - it suggests vm.template might be null which is not possible so let's drop it

Reply:
Now that I think of it... Shouldn't we handle `vm.original_template == null `the same as `vm.original_template.id === emptyGuid`?
Because here if `vm.original_template==null` then `vm.original_template?.id === emptyGuid` is false

Reply:
yeah, it also crossed my mind but that means that if vm.original_template is null and vm.template is not empty, we won't validate the disks as if the vm is not thin/dependent - and that sounds worse.. I'd prefer to drop that check of original-template completely but as @bennyz who wrote this won't be around for couple of weeks, I'd minimize the changes..

Reply:
Agree